lib/workers/pr/index.ts

```diff
@@ -56,9 +56,10 @@ function prepareAssigneesReviewers(
 export function prepareLabels(config: RenovateConfig): string[] {
   const labels = config.labels ?? [];
   const addLabels = config.addLabels ?? [];
-  return [...new Set([...labels, ...addLabels])].map((label) =>
-    template.compile(label, config)
-  );
+  return [...new Set([...labels, ...addLabels])]
+    .filter((el) => el !== null && el !== undefined && el.trim() !== '')
```

Comment:
```suggestion
    .filter(is.nonEmptyString)
```

Reply:
Tests are failing. In this case, `is.nonEmptyString` does not cover a case when string contains multiple white space character.
It still require `trim` e.g. `.filter((el) => is.nonEmptyString(el) && el.trim() !== '')`

Reply:
Thanks for help. Ive updated the logic with `.filter((el) => is.nonEmptyString(el) && !is.emptyStringOrWhitespace(el))` and created a proposal to add `nonEmptyStringOrWhitespace` https://github.com/sindresorhus/is/issues/158